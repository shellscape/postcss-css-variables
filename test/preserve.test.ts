import { fixture, process } from './helpers';

describe('preserve values', async () => {
  test('preserves variables when `preserve` is `true`', async () => {
    const css = await fixture('preserve-value/preserve-variables');
    const result = await process(css, { preserve: true });
    expect(result).toMatchSnapshot();
  });

  test('preserves computed value when `preserve` is `computed`', async () => {
    const css = await fixture('preserve-value/preserve-computed');
    const result = await process(css, { preserve: 'computed' });
    expect(result).toMatchSnapshot();
  });

  test('preserves variables when `preserve` function applies', async () => {
    const css = await fixture('preserve-value/preserve-variables-conditionally');
    const result = await process(css, {
      preserve(declaration) {
        return !(
          declaration.prop.includes('--no-preserve') || declaration.value.includes('--no-preserve')
        );
      }
    });
    expect(result).toMatchSnapshot();
  });
});
