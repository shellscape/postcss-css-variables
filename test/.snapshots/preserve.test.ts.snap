// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preserve values > preserves computed value when \`preserve\` is \`computed\` 1`] = `":root{--color-one:#0000ff;--color-two:#00ff00;--color-three:#00ff00}.foo{prop:before;color:#0000ff}.bar{color:#00ff00;prop:after}.foo-and-bar{prop:before;color:#ff0000;otherprop:after}"`;

exports[`preserve values > preserves variables when \`preserve\` function applies 1`] = `":root{--color-foo:#00ff00;--color-bar:var(--color-two)}.before{prop:before;color:#0000ff}.after{color:#00ff00;color:var(--color-foo);prop:after}.before-and-after{prop:before;color:#ff0000;color:var(--missing,#ff0000);otherprop:after}"`;

exports[`preserve values > preserves variables when \`preserve\` is \`true\` 1`] = `":root{--color-one:#0000ff;--color-two:#00ff00;--color-three:var(--color-two)}.before{prop:before;color:#0000ff;color:var(--color-one)}.after{color:#00ff00;color:var(--color-two);prop:after}.before-and-after{prop:before;color:#ff0000;color:var(--missing,#ff0000);otherprop:after}"`;
